package data.messagedata;

import java.time.LocalDateTime;
import java.util.ArrayList;

import dataenum.UserRole;
import dataenum.ResultMessage;
import javafx.scene.image.ImageView;
import org.junit.*;
import po.messagepo.MessagePO;
import po.userpo.UserPO;

import static org.junit.Assert.*;

/**
 * The class <code>MessageDataServiceImplTest</code> contains tests for the
 * class <code>{@link MessageDataServiceImpl}</code>.
 *
 * @generatedBy CodePro at 18-1-12 下午9:28
 * @author asus
 * @version $Revision: 1.0 $
 */
public class MessageDataServiceImplTest {
	/**
	 * Run the MessageDataServiceImpl() constructor test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testMessageDataServiceImpl_1() throws Exception {

		MessageDataServiceImpl result = new MessageDataServiceImpl();

		// add additional test code here
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_1() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_2() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_3() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_4() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_5() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_6() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_7() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_8() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_9() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_10() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_11() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_12() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_13() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the ArrayList<MessagePO> getMessage(UserPO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessage_14() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		UserPO user = new UserPO("", "", "", UserRole.FINANCIAL_MANAGER, new ImageView());

		ArrayList<MessagePO> result = fixture.getMessage(user);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		assertNotNull(result);
	}

	/**
	 * Run the String getMessageID() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessageID_1() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();

		String result = fixture.getMessageID();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessageID(MessageDataServiceImpl.java:136)
		assertNotNull(result);
	}

	/**
	 * Run the String getMessageID() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessageID_2() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();

		String result = fixture.getMessageID();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessageID(MessageDataServiceImpl.java:136)
		assertNotNull(result);
	}

	/**
	 * Run the String getMessageID() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessageID_3() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();

		String result = fixture.getMessageID();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessageID(MessageDataServiceImpl.java:136)
		assertNotNull(result);
	}

	/**
	 * Run the String getMessageID() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessageID_4() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();

		String result = fixture.getMessageID();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessageID(MessageDataServiceImpl.java:136)
		assertNotNull(result);
	}

	/**
	 * Run the String getMessageID() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessageID_5() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();

		String result = fixture.getMessageID();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessageID(MessageDataServiceImpl.java:136)
		assertNotNull(result);
	}

	/**
	 * Run the String getMessageID() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessageID_6() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();

		String result = fixture.getMessageID();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessageID(MessageDataServiceImpl.java:136)
		assertNotNull(result);
	}

	/**
	 * Run the String getMessageID() method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testGetMessageID_7() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();

		String result = fixture.getMessageID();

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessageID(MessageDataServiceImpl.java:136)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage insert(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testInsert_1() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.insert(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.insert(MessageDataServiceImpl.java:71)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage insert(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testInsert_2() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.insert(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.insert(MessageDataServiceImpl.java:71)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage insert(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testInsert_3() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);

		ResultMessage result = fixture.insert(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.insert(MessageDataServiceImpl.java:71)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage insert(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testInsert_4() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.insert(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.insert(MessageDataServiceImpl.java:71)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage insert(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testInsert_5() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.insert(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.insert(MessageDataServiceImpl.java:71)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage insert(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testInsert_6() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.insert(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.insert(MessageDataServiceImpl.java:71)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage insert(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testInsert_7() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.insert(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.insert(MessageDataServiceImpl.java:71)
		assertNotNull(result);
	}

	/**
	 * Run the void main(String[]) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testMain_1() throws Exception {
		String[] args = new String[] {};

		MessageDataServiceImpl.main(args);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.getMessage(MessageDataServiceImpl.java:104)
		// at
		// data.messagedata.MessageDataServiceImpl.main(MessageDataServiceImpl.java:40)
	}

	/**
	 * Run the ResultMessage save(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testSave_1() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.save(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.save(MessageDataServiceImpl.java:47)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage save(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testSave_2() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.save(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.save(MessageDataServiceImpl.java:47)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage save(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testSave_3() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.save(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.save(MessageDataServiceImpl.java:47)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage save(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testSave_4() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);

		ResultMessage result = fixture.save(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.save(MessageDataServiceImpl.java:47)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage save(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testSave_5() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.save(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.save(MessageDataServiceImpl.java:47)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage save(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testSave_6() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.save(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.save(MessageDataServiceImpl.java:47)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage save(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testSave_7() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.save(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.save(MessageDataServiceImpl.java:47)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage save(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testSave_8() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.save(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.save(MessageDataServiceImpl.java:47)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage update(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testUpdate_1() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.update(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.update(MessageDataServiceImpl.java:86)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage update(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testUpdate_2() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);

		ResultMessage result = fixture.update(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.update(MessageDataServiceImpl.java:86)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage update(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testUpdate_3() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);

		ResultMessage result = fixture.update(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.update(MessageDataServiceImpl.java:86)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage update(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testUpdate_4() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.update(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.update(MessageDataServiceImpl.java:86)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage update(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testUpdate_5() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.update(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.update(MessageDataServiceImpl.java:86)
		assertNotNull(result);
	}

	/**
	 * Run the ResultMessage update(MessagePO) method test.
	 *
	 * @throws Exception
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Test
	public void testUpdate_6() throws Exception {
		MessageDataServiceImpl fixture = new MessageDataServiceImpl();
		MessagePO message = new MessagePO("", "", "", LocalDateTime.MAX, true);
		message.setMessageID("");

		ResultMessage result = fixture.update(message);

		// add additional test code here
		// An unexpected exception was thrown in user code while executing this
		// test:
		// java.lang.NullPointerException
		// at
		// data.messagedata.MessageDataServiceImpl.update(MessageDataServiceImpl.java:86)
		assertNotNull(result);
	}

	/**
	 * Perform pre-test initialization.
	 *
	 * @throws Exception
	 *             if the initialization fails for some reason
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@Before
	public void setUp() throws Exception {
		// add additional set up code here
	}

	/**
	 * Perform post-test clean-up.
	 *
	 * @throws Exception
	 *             if the clean-up fails for some reason
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	@After
	public void tearDown() throws Exception {
		// Add additional tear down code here
	}

	/**
	 * Launch the test.
	 *
	 * @param args
	 *            the command line arguments
	 *
	 * @generatedBy CodePro at 18-1-12 下午9:28
	 */
	public static void main(String[] args) {
		new org.junit.runner.JUnitCore().run(MessageDataServiceImplTest.class);
	}
}